lib/config/configuration.go

```diff
@@ -1113,9 +1113,6 @@ func applyProxyConfig(fc *FileConfig, cfg *servicecfg.Config) error {
 
 	if fc.Proxy.AutomaticUpgradesChannels != nil {
 		cfg.Proxy.AutomaticUpgradesChannels = fc.Proxy.AutomaticUpgradesChannels
-		if err := cfg.Proxy.AutomaticUpgradesChannels.CheckAndSetDefaults(); err != nil {
-			return trace.Wrap(err, "failed to set up automatic update channels")
-		}
```

Comment:
Yes we can start a process with an invalid configuration and fail during proxy service startup. This will interrupt Teleport and cause a shutdown, like other proxy misconfigurations.